org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -1954,7 +1955,7 @@ private void buildTokenLocationContext(InternalCompletionContext context, Scope
 				if (expression.body().sourceStart <= astNode.sourceStart &&
 						astNode.sourceEnd <= expression.body().sourceEnd) {
 					// completion is inside a method body
-					if ((astNodeParent == null || astNodeParent == expression || (astNodeParent instanceof Block)) &&
+					if ((astNodeParent == null || astNodeParent == expression || (astNodeParent instanceof Block) || inParentControlStatement(astNodeParent)) &&
```

Comment:
Yes it will, I'm not sure if we have another `START` constant for this, but there were some tests which was expecting this while completion in side a if condition. Let me try to find those tests.

Reply:
With latest changes this change is not needed.